Default index contributor icon display (#742)

bruceplai opened the pull request:
Closes #742 

* Update display logic to handle showIndexContrib state

* Update checkbox to handle showIndexContrib state

* Update link from home page to have contributor query param

* Update contributors.spec tests

kevinashworth reviewed (changes requested):
Great stuff.

If you also add something like `setValue(2);` at approx line 179, it will open with Affiliated already selected. (And corresponding for other values. It could be renamed to something even better like setTabValue(2) or setValue('affiliated').)

bruceplai commented:
Updated tab state variable to tabValue to be more clear.  Also cleaned up some more code and added cypress tests for /all, /unaffiliated, /affiliated route nav

kevinashworth reviewed (approved):
Great!
